wrapper.ts

```diff
@@ -135,7 +125,24 @@ export class Pty {
       throw err;
     };
 
-    this.read.on('error', handleError);
+    // we need this synthetic eof detector as the pty stream has no way
+    // of distinguishing the program existing vs the data being fully read
+    // this is injected on the rust side after the .wait on the child process
+    // returns
+    // more details: https://github.com/replit/ruspty/pull/93
+    this.read = this.#socket.pipe(new SyntheticEOFDetector());
+    this.write = this.#socket;
+
+    this.#socket.on('error', handleError);
+    this.#socket.once('end', markReadFinished);
+    this.#socket.once('close', handleClose);
+    this.read.once('synthetic-eof', async () => {
```

Comment:
just out of paranoia, is there a way to force-drop the user FD if we get the `close` and never get the synthetic EOF? like after a couple of seconds to avoid FD leaks.